src/core/openthread-core-config.h

```diff
@@ -61,6 +61,7 @@
 #include "config/coap.h"
 #include "config/commissioner.h"
 #include "config/dataset_updater.h"
+#include "config/device_type_check.h"
```

Comment:
Should move this file before other config headers?

Reply:
move it even before `#include OPENTHREAD_PROJECT_CORE_CONFIG_FILE` ?
I think we should detect device as early as possible.